sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -95,59 +91,76 @@ protected ReactorReceiver(AmqpConnection amqpConnection, String entityPath, Rece
+                                final int creditsLeft = receiver.getRemoteCredit();
+
+                                if (creditsLeft > 0) {
+                                    sink.success(message);
+                                    return;
                                 }
-                            }
 
-                            final int creditsLeft = receiver.getRemoteCredit();
+                                final Supplier<Integer> supplier = creditSupplier.get();
+                                final Integer credits = supplier.get();
+
+                                if (credits != null && credits > 0) {
+                                    logger.atVerbose()
+                                        .addKeyValue("credits", credits)
+                                        .log("Adding credits.");
+                                    receiver.flow(credits);
+                                } else {
+                                    logger.atVerbose()
+                                        .addKeyValue("credits", credits)
+                                        .log("There are no credits to add.");
+                                }
 
-                            if (creditsLeft > 0) {
+                                metricsProvider.recordAddCredits(credits == null ? 0 : credits);
                                 sink.success(message);
-                                return;
-                            }
-
-                            final Supplier<Integer> supplier = creditSupplier.get();
-                            final Integer credits = supplier.get();
-
-                            if (credits != null && credits > 0) {
-                                logger.atInfo()
-                                    .addKeyValue("credits", credits)
-                                    .log("Adding credits.");
-                                receiver.flow(credits);
-                            } else {
-                                logger.atVerbose()
-                                    .addKeyValue("credits", credits)
-                                    .log("There are no credits to add.");
-                            }
-
-                            metricsProvider.recordAddCredits(credits == null ? 0 : credits);
-                            sink.success(message);
-                        });
-                    } catch (IOException | RejectedExecutionException e) {
-                        sink.error(e);
+                            });
+                        } catch (IOException | RejectedExecutionException e) {
+                            sink.error(e);
+                        }
+                    });
+                }, 1);
+        } else {
+            this.messagesProcessor = this.handler.getDeliveredMessagesV2()
+                .map(message -> {
+                    // TODO (anu): Check with Liudmila to see if we need isPrefetchedSequenceNumberEnabled() check per message.
```

Comment:
Did we follow up on this?

Reply:
Yep, talked to Liudmila, we don’t expect the meters to be enabled dynamically, which means, we can have the check one time (at the time of Receiver construction) and decide if we should apply `.map` or not. Once less operator :) .